adr/0006-reviewer-responsibilies.adoc

```diff
@@ -0,0 +1,74 @@
+At the same time, some contributions remain unreviewed for a long time because of uncertainty or because of bottleneck.
+
+== Decision Drivers
+
+* Increase clarity and awareness on contributor’s responsibility on reviews and code push
+* Describe the intent and not the current exact state (as tooling is preventing us from having the best solution)
+* Being a perfect solution is _not_ a goal, _better_ than today is.
+* Define ways to measure and improve the time to review.
+
+== Scenarios (optional)
+
+List of scenarios, each described using 2-3 sentences, to capture where the ADR has an impact.
+
+== Considered options
+
+The main alternative is the status quo which roughly is: Guillaume knowing what we are intending to do and some persons talking with Guillaume regularly being sufficiently aware.
+
+== Decision
+
+There are three levels of reviewing.
+
+. Any one can review a Pull Request (PR) and give constructive feedback on different areas (code, documentation, feature experience, security, etc.).
+  This is an optional but recommended step.
+  More eyes means higher quality.
+. A PR is reviewed by a _binding reviewer_ who is fluent in the area being touched: this person is responsible for signing with blood of the usefulness, quality of the code and soundness of the solution.
+  That’s the review of the expert and the most critical one for quality.
+  This review is mandatory.
+.. Unfortunately, we do not always have (designed or effective) experts in all areas so it’s a constant work in progress and might make this step optional in some situations.
+. A PR is reviewed by a committer (push rights) whose responsibilities are:
+.. Do a general review (often from a non expert eye).
+.. Ensure consistency and quality across Quarkus’s codebase (code, concepts, API consistency, documentation).
+.. Detect potential security concerns with an attacker mindset (though not being an actual security professional).
+.. Ensure consistency at the platform level (duplicated or overlapping features, usage integration issues across features etc).
+.. Effectively push the code to the main branch.
+
+=== TODO Open questions to resolve before closing
+
+How is a person moving from casual reviewer to binding reviewer?
+People will tend to request review from binding reviewers leading to a progression problem where casual reviewers will never have the knowledge to be “promoted”.
+We have a few people with no specific area of ownership, that’s specifically acute there.
+
+How can a non binding review see the value of its work considering someone else has to go behind them anyways.
+Feels like double work where the team is very busy.
+
+How can we define SLO for the PR review to help with the _flow_, satisfy contributors while keeping quality high.
+And how to use it as a mechanism to improve and know where to get help?
+The faster the flow, the high an organisation measure in effectiveness, happiness etc.
```

Comment:
Here we go: DORA metrics. 

Let me get this straight—I don't believe in artificial limits. If a PR has been opened for more than n days, let's merge it. This would be terrible and insecure and, in the end, will add even more pressure on the contributors to fix and align these contributions. 

I understand that when you spend time opening a PR, getting staled for n days is not great. It happens, but again, it's not usual — at least not something I'm aware of.  

There are multiple solutions that have been implemented:

1. The lottery: Staled PRs and issues are _re-assigned_ to a set of people (by area), which allows for handling that situation and cleaning up old/out-of-date issues. 
2. Working groups—When an issue is associated with a working group, the main point of contact is responsible for making sure everything is on track. We have periodic reports to capture this. 

Finally, everyone can ping us on Zulip to discuss the issue/PR. As you say, we can be overwhelmed; these discussions let us re-prioritize and bring more people to the table.

Reply:
I did a quick look at the open draft PRs

29 out of 116 are more than a year old, 86 out of 116 are more than a month old (creation time).

And looking at the most recently updated (no longer creation time) and ignoring `draft` and `changes_requested`, there are 41 out of 71 that were updated in the last month and 47 in the last 3 months (24 older).

I expect false positives but by random sampling there were legit drop-off.

Reply:
Can you provide your scripts to detect these? I would not count drafts (now, the question is what do we do with draft PRs once they are stale). Did you run your script only on quarkus "core" repo or on the other repositories too (not speaking of the quarkiverse, but the web site, quarkus-http, quarkus-security, quarkus-spring...).